src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.ByteArray.cs

```diff
@@ -55,7 +55,7 @@ public static partial class JsonSerializer
             JsonSerializerOptions? options = null)
         {
             ValidateInputType(value, inputType);
-            JsonTypeInfo jsonTypeInfo = GetTypeInfo(options, inputType);
+            JsonTypeInfo jsonTypeInfo = GetTypeInfo(options, inputType, fallBackToNearestAncestorType: true);
```

Comment:
cc: @GrabYourPitchforks

Reply:
FWIW this is just falling back to a supported base type for the case of serialization only, so in a way it's the exact opposite of polymorphic serialization.